### PR TITLE
Add period form

### PR DESCRIPTION
**Pull request by usu**
- draft form to add periods
- draft interface between controller and services
- validation exception that contains error messages in array format

### CODE REVIEW

File: module/EcampWeb/src/EcampWeb/Controller/Camp/IndexController.php [hunk 1]
```diff
@@ -81,4 +96,29 @@ public function leaveCampAction()
...
+        $form = new \EcampWeb\Form\PeriodForm($this->getServiceLocator()->get('Doctrine\ORM\EntityManager'));
+        $form->setAttribute('action', $this->url()->fromRoute('web/camp/default', array('camp'=> $this->getCamp(), 'controller'=>'index', 'action' => 'addperiod')));
+
+        if ($this->getRequest()->isPost()) {
+            $form->setData($this->getRequest()->getPost());
+
+            if ( $form->isValid() ) {
+                // save data
+                try {
+                    $period = $this->getServiceLocator()->get('EcampCore\Service\Period')->Create($this->getCamp(), $this->getRequest()->getPost());
```
Comment (pmattmann) [inline module/EcampWeb/src/EcampWeb/Controller/Camp/IndexController.php]: Use here the local method getPeriodService() - not the ServiceLocator

File: module/EcampWeb/src/EcampWeb/Controller/Camp/IndexController.php [hunk 2]
```diff
@@ -81,4 +96,29 @@ public function leaveCampAction()
         );
     }
 
+    public function addPeriodAction()
+    {
+        $form = new \EcampWeb\Form\PeriodForm($this->getServiceLocator()->get('Doctrine\ORM\EntityManager'));
```
Comment (pmattmann) [inline module/EcampWeb/src/EcampWeb/Controller/Camp/IndexController.php]: We have to discuss about this....
I don't like it very much to see the EntityManager here... but I see the motivation...

File: module/EcampWeb/src/EcampWeb/Form/BaseForm.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+use DoctrineModule\Stdlib\Hydrator\DoctrineObject as DoctrineHydrator;
+use Zend\Form\Form;
+use Zend\Form\Element;
+
+abstract class BaseForm extends Form
+{
+    public function __construct($entityManager)
+    {
+        parent::__construct('period-form');
```
Comment (pmattmann) [inline module/EcampWeb/src/EcampWeb/Form/BaseForm.php]: period-form?

File: module/EcampWeb/src/EcampWeb/Controller/Camp/IndexController.php (hunk 1 above)
Comment (usu) [inline module/EcampWeb/src/EcampWeb/Controller/Camp/IndexController.php]: ok

File: module/EcampWeb/src/EcampWeb/Controller/Camp/IndexController.php (hunk 2 above)
Comment (usu) [inline module/EcampWeb/src/EcampWeb/Controller/Camp/IndexController.php]: EntityManager is not needed here. Hydrator is only used in Service, not in Controller. Corrected.

File: module/EcampWeb/src/EcampWeb/Form/BaseForm.php (hunk 3 above)
Comment (usu) [inline module/EcampWeb/src/EcampWeb/Form/BaseForm.php]: copy-paste mistake. corrected.

**Pull request merged**